crates/ruff_python_parser/src/string.rs

```diff
@@ -405,25 +401,21 @@ impl<'a> StringParser<'a> {
             self.next_char();
         }
         if !constant_piece.is_empty() {
-            spec_constructor.push(Expr::from(ast::ExprConstant {
-                value: std::mem::take(&mut constant_piece).into(),
-                kind: None,
+            spec_constructor.push(ast::FStringPart::Literal(ast::PartialString {
+                value: std::mem::take(&mut constant_piece),
                 range: self.range(start_location),
             }));
         }
         Ok(spec_constructor)
     }
 
-    fn parse_fstring(&mut self, nested: u8) -> Result<Vec<Expr>, LexicalError> {
-        use FStringErrorType::{ExpressionNestedTooDeeply, SingleRbrace, UnclosedLbrace};
-
-        if nested >= 2 {
-            return Err(FStringError::new(ExpressionNestedTooDeeply, self.get_pos()).into());
-        }
```

Comment:
I think you moved this because it's only useful for formatted values i.e., the expression between `{` .. `}` or is it any other reason? Isn't `parse_fstring` the entrypoint so does that make any difference?

Reply:
if i recall i think this moved because this function no longer recurses, `parse_formatted_value` does